ols/src/llms/llm_loader.py

```diff
@@ -58,28 +59,47 @@ def __init__(
         self,
         provider: Optional[str] = None,
         model: Optional[str] = None,
-        url: Optional[str] = None,
         params: Optional[dict] = None,
         logger=None,
     ) -> None:
         """Initialize loader using provided provider, model, and other parameters."""
         self.logger = logger if logger is not None else Logger("llm_loader").logger
-        if provider is None:
+        if not provider:
             msg = "Missing provider"
             self.logger.error(msg)
             raise MissingProvider(msg)
         self.provider = provider
-        self.url = url
-        if model is None:
+        if not model:
             msg = "Missing model"
             self.logger.error(msg)
             raise MissingModel(msg)
         self.model = model
+        self.provider_config = self._get_provider_config()
 
         # return empty dictionary if not defined
-        self.llm_params = params if params else {}
+        self.llm_params = params or {}
         self.llm: LLM = self._llm_instance()
 
+    # TODO: refactor after config implementation OLS-89
+    def _get_provider_config(self) -> ProviderConfig:
+        cfg = config.llm_config.providers.get(self.provider)
+        if not cfg:
+            raise UnsupportedProvider()
+
+        model = cfg.models.get(self.model)
+        if not model:
+            # NOTE: temporary condition for WatsonX - implementation is
+            # incompatible with the rest of LLMLoader now
```

Comment:
the WatsonX loader implementation can't handle us throwing a ModelConfigMissingException?

Reply:
It's more like we don't care about it in that case. Currently, we don't have any Watson model in constants or config. It is incompatible in the sense it is not using provider/model values, it is going for envs `WATSON_MODEL` and `WATSON_PROJECT_ID` directly. As it doesn't make sense to use `LLMLoader` with `provider="watsonx"` and `model="something"`, when model is not used, I wanted to separate it.
But I can remove that condition, and leave it for refactoring when someone actually gets credentials to it. WDYT?

Reply:
ok, removed, rebased